### ISSUE TITLE
Make ReplicaExchange into ConditionalSequential EnsembleHops

### ISSUE THREAD

**Issue by dwhswenson** [choderalab/openpathsampling#284]
I think it might be worthwhile to replace the current `ReplicaExchangeMover` with a `ConditionalSequentialMover` consisting of two (unbiased) `EnsembleHopMover`s.

This will give the same results: instead of thinking of replica exchange as a simultaneous swap, we think of it as two independent hops, but the whole move is only accepted if both hops are accepted.

I'd considered this before, but previously didn't see any advantage in it and thought it risked confusion, since that isn't the way people normally think about replica exchange. However, as I'm working on the `MoveStrategy` stuff, I think this might simplify switching between multiple-replica approaches (based on replica exchange) and single-replica (based on ensemble hops). That just becomes a matter of adding/removing the `ConditionalSequentialMover` which makes the replica exchange.

If there's no opposition, I would implement this by making `ReplicaExchangeMover` a subclass of `ConditionalSequentialMover`, so there would still be a separate class for it.

**Comment by jhprinz** [comment]
This is a good idea, but will only work once the "delayed" acceptance works. Once this is done (still working on the PathMove/PMC update) the switch should be very easy and natural. It will also make more complicated replica exchange moves feasible.

Please go ahead.

**Comment by dwhswenson** [comment]
Actually, I don't think it is so necessary for the `MoveStrategy` stuff, now that I'm further into developing that. So I'm back to my original mixed feelings about this, which boil down to:

* **Advantages**: build everything out of the simplest tools (if replica exchange *can* be described as two hops, then why not treat it that way?) It's more structurally and logically satisfying
* **Disadvantages**: no one things of RepEx as a combination of two different moves; everyone thinks of it as one simultaneous move.

Also, you're right that this would require the "delayed acceptance". I hadn't thought of that when I posted this -- definitely a subject we should take up before too long, but since I think I can do the `MoveStrategy` stuff without this, this issue doesn't make the "delayed acceptance" any more urgent than it was otherwise.

**Comment by jhprinz** [comment]
I kind of like the idea to have very simple building blocks and create the rest from it. For everything that is commonly used (like RepEx, Minus, etc...) we will have a convenience mover that does that looks like a single mover, but internally uses the building blocks. This makes it easy to make changes.

At the end, all we want is to build the MoverTree in some way. 

We might even have two views on the tree:
- a simple view that only shows the main blocks
- a complete view that shows all little building blocks

This could actually be connected to the canonical property.

**Comment by dwhswenson** [comment]
When it comes to the visualization, collapsing part of the tree would also be nice for the minus mover. It's a little too large for the tree right now.

This also raises a problem with the current implementation of `canonical`: since it is based on path move changes, you can't tell whether a given mover is canonical before you run the simulation.

In `MoveScheme`s, the moves I consider "canonical" are members of a `group`. (This is NOT the same as `pmc.canonical`, but conceptually, these are the moves I consider as defining "what you're doing during this MC step.") This effectively makes them blocks that can be easily moved around without worrying about the internal structure. The definition of what makes the level of `group` must be made by a human, based on what seems intuitive.

(Also, regarding that visualization: it would be *really* nice to have the mover name instead of the class name. Especially for the named RandomChoiceMovers!)

Anyway, I'm marking this "future". We can do this at some point, but there are more important things to worry about before 1.0.

**Comment by dwhswenson** [comment]
Not necessary.
